core/tee/tee_svc_cryp.c

```diff
@@ -872,6 +899,112 @@ static void op_attr_value_clear(void *attr)
 	*v = 0;
 }
 
+static TEE_Result op_attr_25519_from_user(void *attr, const void *buffer,
+					  size_t size)
+{
+	uint8_t **key = attr;
+
+	if (size != KEY_SIZE_BYTES_25519 || !*key)
+		return TEE_ERROR_SECURITY;
+
+	memcpy(*key, buffer, size);
+
+	return TEE_SUCCESS;
+}
+
+static TEE_Result op_attr_25519_to_user(void *attr,
+					struct ts_session *sess __unused,
+					void *buffer, uint64_t *size)
+{
+	TEE_Result res = TEE_ERROR_GENERIC;
+	uint8_t **key = attr;
+	uint64_t s = 0;
+	uint64_t key_size = (uint64_t)KEY_SIZE_BYTES_25519;
+
+	res = copy_from_user(&s, size, sizeof(s));
+	if (res != TEE_SUCCESS)
+		return res;
+
+	res = copy_to_user(size, &key_size, sizeof(key_size));
+	if (res != TEE_SUCCESS)
+		return res;
+
+	if (s < key_size || !buffer)
+		return TEE_ERROR_SHORT_BUFFER;
+
+	return copy_to_user(buffer, *key, key_size);
+}
+
+static TEE_Result op_attr_25519_to_binary(void *attr, void *data,
+					  size_t data_len, size_t *offs)
+{
+	TEE_Result res = TEE_ERROR_GENERIC;
+	uint8_t **key = attr;
+	size_t next_offs = 0;
+	uint64_t key_size = (uint64_t)KEY_SIZE_BYTES_25519;
+
+	res = op_u32_to_binary_helper(key_size, data, data_len, offs);
+	if (res != TEE_SUCCESS)
+		return res;
+
+	if (ADD_OVERFLOW(*offs, key_size, &next_offs))
+		return TEE_ERROR_OVERFLOW;
+
+	if (data && next_offs <= data_len)
+		memcpy((uint8_t *)data + *offs, *key, key_size);
+	*offs = next_offs;
+
+	return TEE_SUCCESS;
+}
+
+static bool op_attr_25519_from_binary(void *attr, const void *data,
+				      size_t data_len, size_t *offs)
+{
+	uint8_t **key = attr;
+	uint32_t s = 0;
+
+	if (!op_u32_from_binary_helper(&s, data, data_len, offs))
+		return false;
+
+	if (*offs + s > data_len)
+		return false;
+
+	if (s > (uint32_t)KEY_SIZE_BYTES_25519)
+		return false;
+
+	memcpy(*key, (const uint8_t *)data + *offs, s);
+	*offs += s;
+	return true;
+}
+
+static TEE_Result op_attr_25519_from_obj(void *attr, void *src_attr)
+{
+	uint8_t **key = attr;
+	uint8_t **src_key = src_attr;
+
+	if (!*key || !*src_key)
+		return TEE_ERROR_SECURITY;
+
+	memcpy(*key, *src_key, KEY_SIZE_BYTES_25519);
+
+	return TEE_SUCCESS;
+}
+
+static void op_attr_25519_clear(void *attr)
+{
+	uint8_t **key = attr;
+
+	assert(*key);
+
+	memzero_explicit(*key, KEY_SIZE_BYTES_25519);
+}
+
+static void op_attr_25519_free(void *attr)
+{
+	op_attr_25519_clear(attr);
+	free(*(uint8_t**)attr);
```

Comment:
checkpatch complains for a missing space char: `free(*(uint8_t **)attr);`

For consistency, maybe:
```c
	uint8_t **key = attr;

	op_attr_25519_clear(attr);
	free(*key);
```